qiskit/transpiler/target.py

```diff
@@ -1303,3 +1324,29 @@ def target_to_backend_properties(target: Target):
             return BackendProperties.from_dict(properties_dict)
     else:
         return None
+
+
+class FakeTarget(Target):
+    """
+    Pseudo-target class for internal use in the transpilation pipeline.
+    Differently to the :class:`.Target` class, this preudo-target is initialized
+    with a `coupling_map` argument that allows to store connectivity constraints
+    without basis gates.
+    """
+
```

Comment:
Good point with `__len__`, the output is `0` and I will fix it. However, `target is not None` does evaluate to `True` because it's an explicit check for `None`, so at least we can confirm that the pass manager generation logic was working as expected (else it would have been a bigger issue).